x/eibc/keeper/msg_server.go

```diff
@@ -72,6 +72,125 @@ func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrde
 	return &types.MsgFulfillOrderResponse{}, nil
 }
 
+func (m msgServer) FulfillOrderAuthorized(goCtx context.Context, msg *types.MsgFulfillOrderAuthorized) (*types.MsgFulfillOrderAuthorizedResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := ctx.Logger()
+
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	demandOrder, err := m.GetOutstandingOrder(ctx, msg.OrderId)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := m.validateOrder(demandOrder, msg, ctx); err != nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrUnauthorized, err.Error())
+	}
+
+	granterAccount := m.ak.GetAccount(ctx, msg.GetGranterBech32Address())
+	if granterAccount == nil {
+		return nil, types.ErrGranterAddressDoesNotExist
+	}
+
+	// Send the funds from the granterAccount to the eibc packet original recipient
+	err = m.bk.SendCoins(ctx, granterAccount.GetAddress(), demandOrder.GetRecipientBech32Address(), demandOrder.Price)
+	if err != nil {
+		logger.Error("Failed to send price to recipient", "error", err)
+		return nil, err
+	}
+
+	fulfillerAccount := m.ak.GetAccount(ctx, msg.GetFulfillerBech32Address())
+	if fulfillerAccount == nil {
+		return nil, types.ErrFulfillerAddressDoesNotExist
+	}
+
+	feePartReceiver := fulfillerAccount
+	if msg.OperatorAddress != "" {
+		operatorAccount := m.ak.GetAccount(ctx, msg.GetOperatorBech32Address())
+		if operatorAccount == nil {
+			return nil, types.ErrOperatorAddressDoesNotExist
+		}
+		feePartReceiver = operatorAccount
+	}
+
+	fee, _ := sdk.NewDecFromStr(msg.ExpectedFee)
+	fulfillerFee := msg.FulfillerFeePart
+	feePart := fee.Mul(fulfillerFee.Dec).TruncateInt()
+
+	// Send the fee part to the fulfiller/operator
+	err = m.bk.SendCoins(ctx, granterAccount.GetAddress(), feePartReceiver.GetAddress(), sdk.NewCoins(sdk.NewCoin(demandOrder.Price[0].Denom, feePart)))
+	if err != nil {
+		logger.Error("Failed to send fee part to fulfiller", "error", err)
+		return nil, err
+	}
+
+	if err = m.Keeper.SetOrderFulfilled(ctx, demandOrder, fulfillerAccount.GetAddress(), granterAccount.GetAddress()); err != nil {
+		return nil, err
+	}
+
+	if err = uevent.EmitTypedEvent(ctx, demandOrder.GetFulfilledEvent()); err != nil {
+		return nil, fmt.Errorf("emit event: %w", err)
+	}
+
+	return &types.MsgFulfillOrderAuthorizedResponse{}, nil
+}
+
+func (m msgServer) validateOrder(demandOrder *types.DemandOrder, msg *types.MsgFulfillOrderAuthorized, ctx sdk.Context) error {
+	if demandOrder.RollappId != msg.RollappId {
+		return types.ErrRollappIdMismatch
+	}
+
+	if !demandOrder.Price.IsEqual(msg.Price) {
+		return types.ErrPriceMismatch
+	}
+
+	// Check that the fulfiller expected fee is equal to the demand order fee
```

Comment:
```suggestion
	// Check that the fulfiller expected fee is equal to the demand order fee
```

who is the fulfiller in that case? it's confusing. let refrain from using fulfiller in this context please. just liquidity provider and operator.